### PR TITLE
CMake: improve super build usage

### LINKED ISSUE #703: Can't export nor install CMake project depending on absl
Depending on absl's through its CMake buildsystem means that one cannot export a depending project's local buildsystem nor install it because abseil doesn't do either. This prevents further downstream projects from, for example, using `find_package(mylib)` by setting `mylib_DIR` to a build directory of mylib, or from installing mylib (and thus its dependencies) to some local path and using that.

### PR DESCRIPTION
**Pull request by Mizux**
Up to now when integrating abseil-cpp as subproject it is impossible:
- to change option value due to lack of CMP0077
- reuse the absl install rules since there are disabled and we can't change this (use of set() in stead of option())

note: These are blocking google/or-tools so currently we have to maintain a patch.
note: May fix #703

### CODE REVIEW

**Comment by rogeeff** [comment]
Will this restrict usage of our CMake build to new versions of CMake?

**Comment by Mizux** [comment]
TLDR: no

it's more: if you have a recent version of CMake AND you want to use abseil-cpp as subdirectory AND you want to modify option field then you must enable this policy
So here you'll only give more configuration option to user while keeping compatibility with old user cmake

EDIT: this policy has been introduced in CMake 3.13
ref: https://cmake.org/cmake/help/git-stage/manual/cmake-policies.7.html#policies-introduced-by-cmake-3-13
For Googlers: contrary to only have one g3 HEAD, outhere we use versioning scheme so each time CMake want to modify its build system default behaviour (mostly due to mistake) they have to create a policy to introduce the change.

**Comment by rogeeff** [comment]
It appears install is broken now:

+ cmake --build . --target install -- -j
make: *** No rule to make target 'install'.  Stop.

**Comment by Mizux** [comment]
I miss the description field, my bad. I just amend the patch, should be ok now !

**Comment by vslashg** [comment]
Sorry, it looks like our normal pull process fell down somehow.  Your change was incorporated at f624790b7f76ab92fed5ae966abb99a0d455c96f last week, but it lost its link with this PR.  (Our internal source repo serves as the source of truth for Abseil, but our process is supposed to accept PRs from github properly.  That appears not to have happened here.)

I'm closing this PR, since there's nothing left to merge.  Thanks for your contribution, and I'm sorry for the snafu here.
